core/src/test/java/org/eclipse/hono/connection/ConnectionFactoryImplTest.java

```diff
@@ -72,7 +72,7 @@ public void testConnectInvokesHandlerOnfailureToConnect() throws InterruptedExce
         factory.connect(null, null, null, connectionHandler);
```

Comment:
I think that a unit test that needs 2 secs to execute is not very desirable. In this case, I think it is a configuration problem. Can you try to pass in a `ProtonClientOptions` instance to the call to `connect()` and see if this fixes the issue?

```
ProtonClientOptions options = new ProtonClientOptions().setConnectTimeout(100);
factory.connect(options, null, null, connectionHandler);
```